tabular/src/autogluon/tabular/models/widedeep_nn/widedeep_nn_model.py

```diff
@@ -0,0 +1,289 @@
+        y_norm, y_val_norm = self.y_scaler.fit_transform(y, y_val)
+
+        embed_cols, for_transformer = self.__get_embedding_columns_metadata(X, cat_cols)
+        self._tab_preprocessor = TabPreprocessor(
+            embed_cols=embed_cols,
+            continuous_cols=cont_cols,
+            for_transformer=for_transformer
+        )
+
+        X_tab = self._tab_preprocessor.fit_transform(X)
+        X_train = {'X_tab': X_tab, 'target': y_norm.values}
+        if X_val is not None and y_val is not None:
+            X_valid = {'X_tab': self._tab_preprocessor.transform(X_val), 'target': y_val_norm.values}
+            val_split = None
+        else:
+            X_valid = None
+            val_split = 0.1
+
+        return X_train, X_valid, cont_cols, embed_cols, val_split
+
+    def __get_embedding_columns_metadata(self, X, cat_cols):
+        for_transformer = self.params['type'] in ['tab_transformer', 'ft_transformer', 'tab_perciever', 'tab_fastformer']
+        if for_transformer:
+            embed_cols = cat_cols
+        else:
+            embed_cols = []
+            for cat_feat in cat_cols:
+                num_categories = len(X[cat_feat].cat.categories)
+                embed_cols.append((cat_feat, min(600, round(1.6 * num_categories ** 0.56))))
+        if len(embed_cols) == 0:
+            embed_cols = None
+            for_transformer = False
+        return embed_cols, for_transformer
+
+    @staticmethod
+    def __construct_wide_deep_model(model_type, column_idx, embed_input, continuous_cols, pred_dim, **model_args):
+        from pytorch_widedeep.models import TabMlp, WideDeep, TabResnet, SAINT, TabTransformer, FTTransformer, TabPerceiver, ContextAttentionMLP, SelfAttentionMLP, TabNet, TabFastFormer
+
+        model_cls = dict(
+            tab_mlp=TabMlp,
+            context_attention_mlp=ContextAttentionMLP,
+            self_attention_mlp=SelfAttentionMLP,
+            tabresnet=TabResnet,
+            tabnet=TabNet,
+            SAINT=SAINT,
+            tab_transformer=TabTransformer,
+            ft_transformer=FTTransformer,
+            tab_fastformer=TabFastFormer,
+            tab_perciever=TabPerceiver,
+        ).get(model_type, None)
+
+        if model_cls is None:
+            raise ValueError(f'Unknown model type {model_type}')
+
+        model = model_cls(
+            column_idx=column_idx,
+            cat_embed_input=embed_input,
+            continuous_cols=continuous_cols,
+            **model_args
+        )
+        model = WideDeep(deeptabular=model, pred_dim=pred_dim)
+        return model
+
+    def _set_default_params(self):
+        """ Specifies hyperparameter values to use by default """
+        default_params = get_param_baseline(self.problem_type)
+        for param, val in default_params.items():
+            self._set_default_param_value(param, val)
+
+    def _get_default_searchspace(self):
+        return get_default_searchspace(self.problem_type, num_classes=None)
```

Comment:
if num_classes don't matter, then remove it from parameters in function.

Reply:
fixed